packages/react/src/select/popup/SelectPopup.tsx

```diff
@@ -88,6 +90,7 @@ const SelectPopup = React.forwardRef(function SelectPopup(
         modal={false}
         disabled={!mounted}
         visuallyHiddenDismiss={modal ? 'Dismiss popup' : undefined}
+        initialFocus={selectedIndex === null ? popupRef : undefined}
         order={['content', 'reference']}
       >
```

Comment:
Removing `order` also fixes it, which was added with modality changes, but not sure if it's truly necessary here @michaldudak